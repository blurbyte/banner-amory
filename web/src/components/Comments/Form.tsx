import styled from 'styled-components';
import { transparentize } from 'polished';

import { Color } from '@styles/color';

const Form = styled.form`
  display: grid;
  grid-template-columns: 15rem 1fr 25rem;
  border-bottom: 1px solid ${transparentize(0.5, Color.Blue)};
`;

export default Form;
