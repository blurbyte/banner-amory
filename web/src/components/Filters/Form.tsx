import styled from 'styled-components';

type FormProps = {
  isExpanded?: boolean;
};

const Form = styled.form<FormProps>`
  overflow: hidden;
  height: ${props => (props.isExpanded ? '12rem' : 0)};
  transition: height 0.25s ease-in-out;
  padding-left: 10rem;
  display: flex;
  flex-wrap: wrap;
  align-content: flex-start;
`;

export default Form;
