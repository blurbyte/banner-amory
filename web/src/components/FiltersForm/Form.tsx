import styled from 'styled-components';

type FormProps = {
  isExpanded?: boolean;
};

const Form = styled.form<FormProps>`
  overflow: hidden;
  height: ${props => (props.isExpanded ? '14rem' : 0)};
  transition: height 0.25s ease-in-out;
  padding-left: 10rem;
`;

export default Form;
