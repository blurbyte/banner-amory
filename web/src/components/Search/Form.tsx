import styled from 'styled-components';

const Form = styled.form`
  position: relative;
  z-index: 9999;
`;

export default Form;
