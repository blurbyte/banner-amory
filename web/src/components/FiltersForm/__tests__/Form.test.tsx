import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import Form from '../Form';

test('renders correctly', () => {
  const component = renderer.create(<Form />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders correctly when it is expanded', () => {
  const component = renderer.create(<Form isExpanded />);
  expect(component.toJSON()).toMatchSnapshot();
});
