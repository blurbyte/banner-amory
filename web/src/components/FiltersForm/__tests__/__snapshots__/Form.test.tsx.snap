// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  overflow: hidden;
  height: 0;
  -webkit-transition: height 0.25s ease-in-out;
  transition: height 0.25s ease-in-out;
  padding-left: 10rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

<form
  className="c0"
/>
`;

exports[`renders correctly when it is expanded 1`] = `
.c0 {
  overflow: hidden;
  height: 12rem;
  -webkit-transition: height 0.25s ease-in-out;
  transition: height 0.25s ease-in-out;
  padding-left: 10rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

<form
  className="c0"
/>
`;
